Comprensión de los Datos

# Exploración básica de Pandas — Diabetes
# Nombre: Camille Yamilé Caltenco García  
# Matrícula: A00842805  

In [2]:
#importa librerías
import pandas as pd

# Descripción de Variables


Pclass Passenger Class (1 = 1st; 2 = 2nd; 3 = 3rd): Categórica Nominal 
survival Survival (0 = No; 1 = Yes)  
name Name  
sex Sex  
age Age  
sibsp Number of Siblings/Spouses Aboard  
parch Number of Parents/Children Aboard  
ticket Ticket Number  
fare Passenger Fare (British pound)  
cabin Cabin  
embarked Port of Embarkation (C = Cherbourg; Q = Queenstown; S = Southampton)  
boat Lifeboat  
body Body Identification Number  
home.dest Home/Destination

**Ejemplo:** Crear un objeto DataFrame con base en un archivo .csv

In [3]:
#lee archivo csv
df = pd.read_csv("diabetes.csv")


In [4]:
#Usa función shape para revisar el total de renglones y columnas
df.shape

(768, 9)

In [5]:
#Revisa los primeros 5 renglones del dataset usando la función head()
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [6]:
#Revisa los últimos 5 renglones del dataset usando la función tail()
df.tail() #default 5

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0

In [7]:
#Revisa la información mas completa del conjunto de datos usando la función info()
#Muestra el total de datos, las columnas y su tipo correspondiente, dice si contiene nulos o no
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


los atributos age y cabin tienen valores No nulos.

In [8]:
#revisa cuántos valores únicos tiene cada atributo del archivo usando la función nunique()
df.nunique()

Pregnancies                  17
Glucose                     136
BloodPressure                47
SkinThickness                51
Insulin                     186
BMI                         248
DiabetesPedigreeFunction    517
Age                          52
Outcome                       2
dtype: int64

## Exploración de Datos

In [9]:
#utiliza la función describe() para obtener estadística básica. se puede incluir -0
df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [48]:
df[['DiabetesPedigreeFunction', 'Age', 'Outcome']].describe()

DiabetesPedigreeFunction         Age     Outcome
count                768.000000  768.000000  768.000000
mean                   0.471876   33.240885    0.348958
std                    0.331329   11.760232    0.476951
min                    0.078000   21.000000    0.000000
25%                    0.243750   24.000000    0.000000
50%                    0.372500   29.000000    0.000000
75%                    0.626250   41.000000    1.000000
max                    2.420000   81.000000    1.000000

### Análisis e Interpretación

- **DiabetesPedigreeFunction:**  
  La media (~0.47) y mediana (~0.37) indican que la mayoría de los pacientes tienen un riesgo familiar bajo.  
  La desviación estándar (~0.33) muestra una variabilidad moderada.  

- **Age:**  
  La media (~33 años) sugiere que la mayoría de los pacientes son adultos jóvenes.  
  La desviación estándar (~11 años) indica que hay personas desde los 20 hasta más de 70 años.  

- **Outcome:**  
  El valor promedio (~0.35) muestra que aproximadamente el 35% de los pacientes del conjunto tienen diabetes.  
  Esto sugiere que la mayoría de los registros son de personas sin diabetes.


In [13]:
#Revisa Valores nulos con funcion isnull().sum()
df.isnull().sum()


Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [16]:
#Revisar valores únicos por columna usando función unique(): nombre-columna.unique()
df.Age.unique()

array([50, 31, 32, 21, 33, 30, 26, 29, 53, 54, 34, 57, 59, 51, 27, 41, 43,
       22, 38, 60, 28, 45, 35, 46, 56, 37, 48, 40, 25, 24, 58, 42, 44, 39,
       36, 23, 61, 69, 62, 55, 65, 47, 52, 66, 49, 63, 67, 72, 81, 64, 70,
       68])

In [17]:
df.DiabetesPedigreeFunction.unique()

array([0.627, 0.351, 0.672, 0.167, 2.288, 0.201, 0.248, 0.134, 0.158,
       0.232, 0.191, 0.537, 1.441, 0.398, 0.587, 0.484, 0.551, 0.254,
       0.183, 0.529, 0.704, 0.388, 0.451, 0.263, 0.205, 0.257, 0.487,
       0.245, 0.337, 0.546, 0.851, 0.267, 0.188, 0.512, 0.966, 0.42 ,
       0.665, 0.503, 1.39 , 0.271, 0.696, 0.235, 0.721, 0.294, 1.893,
       0.564, 0.586, 0.344, 0.305, 0.491, 0.526, 0.342, 0.467, 0.718,
       0.962, 1.781, 0.173, 0.304, 0.27 , 0.699, 0.258, 0.203, 0.855,
       0.845, 0.334, 0.189, 0.867, 0.411, 0.583, 0.231, 0.396, 0.14 ,
       0.391, 0.37 , 0.307, 0.102, 0.767, 0.237, 0.227, 0.698, 0.178,
       0.324, 0.153, 0.165, 0.443, 0.261, 0.277, 0.761, 0.255, 0.13 ,
       0.323, 0.356, 0.325, 1.222, 0.179, 0.262, 0.283, 0.93 , 0.801,
       0.207, 0.287, 0.336, 0.247, 0.199, 0.543, 0.192, 0.588, 0.539,
       0.22 , 0.654, 0.223, 0.759, 0.26 , 0.404, 0.186, 0.278, 0.496,
       0.452, 0.403, 0.741, 0.361, 1.114, 0.457, 0.647, 0.088, 0.597,
       0.532, 0.703,

In [20]:
df.Outcome.unique()

array([1, 0])

## Variables Cuantitativas

### Medidas de tendencia central


In [18]:
#Edad
#Se puede obtener la media, mediana y moda para
mean_age = df['Age'].mean()
median_age = df['Age'].median()
mode_age = df['Age'].mode()
print("Mean_age:",mean_age)
print("Median_age:",median_age)
print("Mode_age:",mode_age)

Mean_age: 33.240885416666664
Median_age: 29.0
Mode_age: 0    22
Name: Age, dtype: int64


In [21]:
#Outcome
#Se puede obtener la media, mediana y moda para
mean_Outcome = df['Outcome'].mean()
median_Outcome = df['Outcome'].median()
mode_Outcome = df['Outcome'].mode()
print("Mean_Outcome:",mean_Outcome)
print("Median_Outcome:",median_Outcome)
print("Mode_Outcome:",mode_Outcome)

Mean_Outcome: 0.3489583333333333
Median_Outcome: 0.0
Mode_Outcome: 0    0
Name: Outcome, dtype: int64


In [22]:
#DiabetesPedigreeFunction
#Se puede obtener la media, mediana y moda para
mean_DiabetesPedigreeFunction = df['DiabetesPedigreeFunction'].mean()
median_DiabetesPedigreeFunction = df['DiabetesPedigreeFunction'].median()
mode_DiabetesPedigreeFunction = df['DiabetesPedigreeFunction'].mode()
print("Mean_DiabetesPedigreeFunction:",mean_DiabetesPedigreeFunction)
print("Median_DiabetesPedigreeFunction:",median_DiabetesPedigreeFunction)
print("Mode_DiabetesPedigreeFunction:",mode_DiabetesPedigreeFunction)

Mean_DiabetesPedigreeFunction: 0.47187630208333325
Median_DiabetesPedigreeFunction: 0.3725
Mode_DiabetesPedigreeFunction: 0    0.254
1    0.258
Name: DiabetesPedigreeFunction, dtype: float64


### Descripción de Variables Analizadas

**DiabetesPedigreeFunction:**  
Tipo: Cuantitativa continua  
Representa una función que estima la probabilidad de tener diabetes según los antecedentes familiares.  
Rango: de 0.078 a 2.42  

**Age:**  
Tipo: Cuantitativa discreta  
Representa la edad de las personas en años.  
Rango: de 21 a 81  

**Outcome:**  
Tipo: Categórica binaria  
Indica si el paciente tiene diabetes (1) o no (0).  
Rango: 0 o 1


# Variables Categóricas

In [23]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.Age.value_counts()

Age
22    72
21    63
25    48
24    46
23    38
28    35
26    33
27    32
29    29
31    24
41    22
30    21
37    19
42    18
33    17
36    16
38    16
32    16
45    15
34    14
46    13
40    13
43    13
39    12
35    10
44     8
50     8
51     8
52     8
58     7
54     6
47     6
49     5
60     5
53     5
57     5
48     5
63     4
66     4
55     4
62     4
59     3
56     3
65     3
67     3
61     2
69     2
72     1
81     1
64     1
70     1
68     1
Name: count, dtype: int64

In [24]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.DiabetesPedigreeFunction.value_counts()

DiabetesPedigreeFunction
0.258    6
0.254    6
0.268    5
0.207    5
0.261    5
        ..
0.427    1
1.213    1
0.329    1
1.318    1
0.933    1
Name: count, Length: 517, dtype: int64

In [25]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.Outcome.value_counts()

Outcome
0    500
1    268
Name: count, dtype: int64

## Consulta

In [26]:
# df.iloc[i]: Accede a la fila en la posición i.
# Acceder a la primera fila
df.iloc[1]

Pregnancies                  1.000
Glucose                     85.000
BloodPressure               66.000
SkinThickness               29.000
Insulin                      0.000
BMI                         26.600
DiabetesPedigreeFunction     0.351
Age                         31.000
Outcome                      0.000
Name: 1, dtype: float64

In [27]:
# Acceder a las dos primeras filas
df.iloc[1:3]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   

   DiabetesPedigreeFunction  Age  Outcome  
1                     0.351   31        0  
2                     0.672   32        1

In [49]:
# 1. Promedio de la función de parentesco en personas con y sin diabetes
df.groupby('Outcome')['DiabetesPedigreeFunction'].mean()

Outcome
0    0.429734
1    0.550500
Name: DiabetesPedigreeFunction, dtype: float64

In [50]:
# 2. Edad promedio de las personas con diabetes
df[df['Outcome'] == 1]['Age'].mean()

np.float64(37.06716417910448)

In [52]:
# 3. Porcentaje de personas mayores de 50 años con diabetes
(df[(df['Age'] > 50) & (df['Outcome'] == 1)].shape[0] / df[df['Age'] > 50].shape[0]) * 100

46.913580246913575

In [53]:
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

In [ ]:
### Conclusiones

El análisis muestra que el riesgo familiar (DiabetesPedigreeFunction) se relaciona con la presencia de diabetes, aunque no es el único factor.  
La mayoría de los pacientes son adultos jóvenes, pero los casos aumentan en edades mayores.  
En general, solo un tercio de los pacientes presenta diabetes, lo que indica que el conjunto de datos está balanceado hacia personas sin la enfermedad.